lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1472,4 +1472,106 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 0));
+        for (int i = 0; i < shapeSizes[0]; i++) {
+          Value selectIndex = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(),
+              rewriter.getIntegerAttr(rewriter.getIntegerType(64), i));
+          Value extract = rewriter.create<Torch::AtenSelectIntOp>(
+              binder.getLoc(), selectResultType, shape, zero, selectIndex);
+          Value dim = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), extract);
+          dimList.push_back(dim);
+        }
+        Value dimValueList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            dimList);
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+
+        // Get fill_value if it is present.
+        // Assumption : resultDType and value attr type match.
+        Value value_const;
+        auto attr = binder.op->getAttr("torch.onnx.value");
+        auto resultDType = resultType.getDtype();
+        struct FillValueParam {
+          enum { FLOAT, INT } dType; // Add more types as needed
+          union {
+            float floatVal;
+            int64_t intVal;
+          };
+        };
+
+        // Extract the fill value and dtype
+        // ONNX requires value attr to be a tensor
+        if (!attr) {
+          attr = DenseElementsAttr::get(
+              resultType.toBuiltinTensor().clone(resultDType),
+              rewriter.getFloatAttr(resultDType, 0.0));
+        }
+        if (!isa<DenseElementsAttr>(attr)) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "`value` attr needs to be a tensor.");
+        }
+
+        auto denseAttr = attr.dyn_cast_or_null<DenseElementsAttr>();
+        auto denseAttrEleType = denseAttr.getElementType();
+        if (!isa<FloatType, IntegerType>(denseAttrEleType)) {
+          return rewriter.notifyMatchFailure(
+              binder.op,
+              "`value` attr tensor only supports types int and float for now.");
+        }
+
+        // Store the value and dtype info
+        FillValueParam fvp;
+        if (denseAttrEleType.isa<IntegerType>()) {
+          fvp.dType = FillValueParam::INT;
+          fvp.intVal = denseAttr.getSplatValue<IntegerAttr>().getSInt();
+        }
+        if (denseAttrEleType.isa<FloatType>()) {
+          fvp.dType = FillValueParam::FLOAT;
+          fvp.floatVal =
+              denseAttr.getSplatValue<FloatAttr>().getValue().convertToFloat();
+        }
+
+        // Create the constant Op for value
+        switch (fvp.dType) {
+        case FillValueParam::INT:
+          value_const = rewriter.create<Torch::ConstantIntOp>(
```

Comment:
Why not move these instantiations into the `if` statements above? We could forgo the entire `switch` and `struct` types you defined.

Reply:
Done